master/pkg/checkpoints/s3/s3checkpoints.go

```diff
@@ -35,21 +36,27 @@ func (w *seqWriterAt) WriteAt(p []byte, off int64) (int, error) {
 // GetS3BucketRegion returns the region name of the specified bucket.
 // It does so by making an API call to AWS.
 func GetS3BucketRegion(ctx context.Context, bucket string) (string, error) {
-	sess, err := session.NewSession(&aws.Config{
-		Region: aws.String("us-west-2"),
-	})
+	// TODO this won't work on non aws s3 APIs.
```

Comment:
unfortunately, if this isn't going to work within minio, it is sort of a non-starter. though, did this work before with minio, anyway?

can we have add an optional region field to `checkpoint_storage.type == s3` instead? trying to think about alternatives.

Reply:
I don't know how it could have worked with minio, we don't have the ```endpoint_url``` here

The caller can check ```endpoint_url``` to check if it is a non aws bucket. 

https://github.com/minio/minio/discussions/15063 minio seems to want us-east-1 so maybe we just default to that if ```endpoint_url``` is not empty and not s3?

Reply:
though I do think this is pretty separate from this change and something we would also want to test in CI

Reply:
created https://hpe-aiatscale.atlassian.net/browse/DET-10035

do you know if minio is the only s3 clone people use? Or if there are others we should probably test with and make sure they work?